File: update_nova_page.py
with open('src/app/dashboard/solicitante/nova/page.tsx', 'r', encoding='utf-8') as f:
    content = f.read()

old_dept = "const departments = await prisma.department.findMany({ orderBy: { name: 'asc' } })"

new_dept = """let departments: any[] = []
  if (user.role === 'ADMIN' || user.role === 'AUTORIZADOR' || user.role === 'COMPRADOR') {
    departments = await prisma.department.findMany({ orderBy: { name: 'asc' } })
  } else {
    // Solicitante vê apenas seus próprios setores
    if (user.department) departments.push(user.department)
    if (user.additionalDepartments) {
      user.additionalDepartments.forEach((d: any) => {
        if (!departments.find((existing: any) => existing.id === d.id)) {
          departments.push(d)
        }
      })
    }
    departments.sort((a, b) => a.name.localeCompare(b.name))
  }"""

content = content.replace(old_dept, new_dept)

with open('src/app/dashboard/solicitante/nova/page.tsx', 'w', encoding='utf-8') as f:
    f.write(content)

File: update_nova_page.js
const fs = require('fs');
let content = fs.readFileSync('src/app/dashboard/solicitante/nova/page.tsx', 'utf8');

// Load all users
const loadTarget = "const backLink = (user.role === 'COMPRADOR' || user.role === 'AUTORIZADOR') ? '/dashboard/comprador' : '/dashboard/solicitante'";
const loadReplace = "const allUsers = await prisma.user.findMany({ select: { id: true, name: true, department: true }, orderBy: { name: 'asc' } })\n\n  const backLink = (user.role === 'COMPRADOR' || user.role === 'AUTORIZADOR') ? '/dashboard/comprador' : '/dashboard/solicitante'";
content = content.replace(loadTarget, loadReplace);

// Pass to RequestForm
const formTarget = "<RequestForm user={user} groups={groups} targetUsers={targetUsers} isComprador={user.role === 'COMPRADOR' || user.role === 'AUTORIZADOR'} departments={departments} />";
const formReplace = "<RequestForm user={user} groups={groups} targetUsers={targetUsers} allUsers={allUsers} isComprador={user.role === 'COMPRADOR' || user.role === 'AUTORIZADOR'} departments={departments} />";
content = content.replace(formTarget, formReplace);

fs.writeFileSync('src/app/dashboard/solicitante/nova/page.tsx', content);